### DISCUSSION TITLE
How to call read-only contract functions before user connects wallet?

### DISCUSSION THREAD

**Discussion by RogerPodacter** [wevm/wagmi#1039] [Q&A] [answered]
It is nice to show the user a partially-functional app before they connect their wallet. However, with the default Wagmi setup I can't seem to call any contract functions through `useContractRead` without the user having connected their wallet.

Is there a way to accomplish this?

**Comment by tmm** [comment, marked as answer]
`useContractRead` should work fine without a wallet connected. Have an example to look at?

**Reply by RogerPodacter** [reply]
Ah very weird; I'm using this config:

```javascript
const { chains, provider } = configureChains(allChains, [
  alchemyProvider({ apiKey: alchemyId }),
  publicProvider(),
])
```

And for some reason the provider that's generated is correctly on Goerli when a wallet is connected and incorrectly on Mainnet when a wallet is not. I will continue investigating!

**Reply by RogerPodacter** [reply]
Ok I think I fixed it with this!

```javascript
function findChain(id: string) {
  return allChains.find((x) => x.id === +id)
}

const chain = findChain(process.env.NEXT_PUBLIC_CHAIN_ID || "1")

// Set up chains
const { chains, provider } = configureChains([chain || allChains[0]], [
  alchemyProvider({ apiKey: alchemyId }),
  publicProvider(),
])
```
